asab/web/rest/json.py

```diff
@@ -142,26 +142,40 @@ def json_schema_handler(schema, *_args, **_kwargs):
 		'key1': {'type': 'string'},
 		'key2': {'type': 'number'},
 	}})
-	async def login(self, request):
+	async def login(self, request, *, validated_data):
 		...
-	'''
+
+	Works for `application/json`, `application/x-www-form-urlencoded` and `multipart/form-data` post requests
+	"""
 
 	def decorator(func):
 
+		validate = fastjsonschema.compile(schema)
+
 		async def validator(*args, **kwargs):
 			# Initializing fastjsonschema.compile method and generating
 			# the validation function for validating JSON schema
-			validate = fastjsonschema.compile(schema)
 			request = args[-1]
-			data = await request.text()
+			if request.content_type == 'application/json':
+				data = await request.json()
+			elif request.content_type in ('', 'application/x-www-form-urlencoded', 'multipart/form-data'):
+				multi_dict = await request.post()
+				data = {k: v for k, v in multi_dict.items()}
+			else:
+				Lex.warning(f"Unsupported content-type {request.content_type} for request {request}")
+				raise aiohttp.web.HTTPBadRequest(reason=f"Unsupported content-type {request.content_type}")
 			# Checking the validation on JSON data set
 			try:
-				validate(json.loads(data))
-				kwargs['valid'] = json.loads(data)
+				validate(data)
+				kwargs['validated_data'] = data
```

Comment:
`jsonobj` is my best proposal for the name.

Reply:
I'll leave it as `json_data` as it is in #79